### PR TITLE
Replace MBEDTLS_ECP_HAVE_CURVE* with PSA_WANT counterparts

### LINKED ISSUE #9333: Replace `MBEDTLS_ECP_HAVE_CURVE*`  with its `PSA_WANT` counterparts
Replace these macros:
- `MBEDTLS_ECP_HAVE_CURVE25519` to `PSA_WANT_ECC_MONTGOMERY_255`
- `MBEDTLS_ECP_HAVE_CURVE448` to `PSA_WANT_ECC_MONTGOMERY_448`

The macros should be replaced in all files except `mbedtls_config.h`, `cneck_config.h` and `config_adjust_*.h`.

Ensure that the test are run in the same way.

### PR DESCRIPTION
**Pull request by eleuzi01**
## Description

Replace  `MBEDTLS_ECP_HAVE_CURVE25519` with `PSA_WANT_ECC_MONTGOMERY_255` and
`MBEDTLS_ECP_HAVE_CURVE448` with `PSA_WANT_ECC_MONTGOMERY_448`

Resolves #9333 

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required, internal changes
- [x] **3.6 backport** not required, needed for 4.0
- [x] **2.28 backport** not required, needed for 4.0
- [x] **tests** not required

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

**Review by sezrab — DISMISSED** [review]
LGTM

File: tf-psa-crypto/drivers/builtin/src/pk_internal.h [hunk 1]
```diff
@@ -114,9 +114,9 @@ static inline mbedtls_ecp_group_id mbedtls_pk_get_ec_group_id(const mbedtls_pk_c
 }
 
 /* Helper for Montgomery curves */
-#if defined(MBEDTLS_ECP_HAVE_CURVE25519) || defined(MBEDTLS_ECP_HAVE_CURVE448)
+#if defined(PSA_WANT_ECC_MONTGOMERY_255) || defined(PSA_WANT_ECC_MONTGOMERY_448)
 #define MBEDTLS_PK_HAVE_RFC8410_CURVES
-#endif /* MBEDTLS_ECP_HAVE_CURVE25519 || MBEDTLS_ECP_DP_CURVE448 */
+#endif /* PSA_WANT_ECC_MONTGOMERY_255 || MBEDTLS_ECP_DP_CURVE448 */
```
Comment (gabor-mezei-arm) [inline tf-psa-crypto/drivers/builtin/src/pk_internal.h:119]:
```suggestion
#endif /* PSA_WANT_ECC_MONTGOMERY_255 || PSA_WANT_ECC_MONTGOMERY_448 */
```

It seems it was not updated to the legacy macro when it was introduced

**Review by gabor-mezei-arm — CHANGES_REQUESTED** [review]

File: tf-psa-crypto/drivers/builtin/src/pk_internal.h (hunk 1 above)
Comment (eleuzi01) [reply tf-psa-crypto/drivers/builtin/src/pk_internal.h]: Done, thanks for noticing

Comment (eleuzi01) [reply tf-psa-crypto/drivers/builtin/src/pk_internal.h]: Done, thanks for noticing

**Review by gabor-mezei-arm — APPROVED** [review]
LGTM

**Pull request merged**